code/game/objects/structures/displaycase.dm

```diff
@@ -47,27 +89,65 @@
 	return
 
 /obj/structure/displaycase/update_icon()
-	if(src.destroyed)
-		src.icon_state = "glassboxb[src.occupied]"
-	else
-		src.icon_state = "glassbox[src.occupied]"
+	underlays.Cut()
+	if(displayed_item)
+		var/image/I
+		I = image(displayed_item.icon, displayed_item.icon_state)
+		I.pixel_x = item_x_offset
+		I.pixel_y = item_y_offset
+		underlays += I
+	if(destroyed)
+		icon_state = "glassboxb0"
+	if(open && !destroyed)
+		icon_state = "glassbox_open"
+	if(!open && !destroyed)
+		icon_state = "glassbox"
 	return
 
 
 /obj/structure/displaycase/attackby(obj/item/W as obj, mob/user as mob)
-	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
-	src.health -= W.force
-	src.healthcheck()
-	..()
-	return
+	if(istype(W,/obj/item/card/id))
+		var/obj/item/card/id/C = W
+		if(src.access_needed in C.access && !destroyed)
+			open_close(user)
+			update_icon()
+	if(istype(W,/obj/item))
+		if(open && !displayed_item && !destroyed)
+			user.drop_from_inventory(W,src)
+			W.forceMove(src)
+			displayed_item = W
+			update_icon()
+	if(istype(W, /obj/item/stack/material) && W.get_material_name() == "glass")
+		var/obj/item/stack/G = W
+		if(health >= maxhealth && !destroyed)
+			to_chat(user, "<span class='notice'>The case cannot be repaired further!.</span>")
+			return
+		if(health <= maxhealth && !destroyed)
+			if(G.use(2))
+				playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+				to_chat(user, "<span class='notice'>You places the glass on the [src].</span>")
+				health = max(health+(maxhealth/5), maxhealth)
+			else
+				to_chat(user, "<span class='warning'>You need 2 sheets of glass to repair this!.</span>")
+	else if(user.a_intent == I_HURT)
+		user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+		src.health -= W.force
+		src.healthcheck()
+		..()
+		return
 
 /obj/structure/displaycase/attack_hand(mob/user as mob)
-	if (src.destroyed && src.occupied)
-		new /obj/item/gun/energy/captain( src.loc )
-		to_chat(user, "<span class='notice'>You deactivate the hover field built into the case.</span>")
-		src.occupied = 0
+	if (src.open && displayed_item)
+		to_chat(user, "<span class='notice'>You remove the [displayed_item.name].</span>")
+		displayed_item.forceMove(src.loc)
+		displayed_item = null
 		src.add_fingerprint(user)
 		update_icon()
+		if(alarm)
+			if(delayed_alarm)
+				addtimer(CALLBACK(src, .proc/tripalarm, security_level), alarm_delay SECONDS)
```

Comment:
Yeah, uh, don't put the SECONDS there. It should just be alarm_delay. And the seconds should be defined in the alarm_delay declaration.